e2e/cypress/integration/channel_sidebar/category_collapsing_unread_filter_spec.js

```diff
@@ -0,0 +1,291 @@
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS) i').should('be.visible').should('not.have.class', 'icon-rotate-minus-90');
+
+        // # Click on PUBLIC CHANNELS
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS)').should('be.visible').click();
+
+        // * Verify that the category appears collapsed after refresh
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS) i').should('have.class', 'icon-rotate-minus-90');
+
+        // Wait for state to settle
+        cy.wait(1000);
+
+        // # Reload the page
+        cy.reload();
+
+        // * Verify that the category still appears collapsed after refresh
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS) i').should('have.class', 'icon-rotate-minus-90');
+
+        // # Click on PUBLIC CHANNELS
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS)').should('be.visible').click();
+
+        // * Verify that the category appears not collapsed after refresh
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS) i').should('not.have.class', 'icon-rotate-minus-90');
+
+        // Wait for state to settle
+        cy.wait(1000);
+
+        // # Reload the page
+        cy.reload();
+
+        // * Verify that the category still appears not collapsed after refresh
+        cy.get('.SidebarChannelGroupHeader:contains(PUBLIC CHANNELS) i').should('not.have.class', 'icon-rotate-minus-90');
+    });
+
+    it('should change the text state when the unread filter changes state', () => {
+        // # Start with a new team
+        const teamName = `team-${getRandomInt(999999)}`;
+        cy.createNewTeam(teamName, teamName);
+
+        // * Verify that we've switched to the new team and are on Town Square
+        cy.get('#headerTeamName').should('contain', teamName);
+        cy.url().should('include', `/${teamName}/channels/town-square`);
+
+        // * Verify that the unread filter is in all channels state
+        cy.get('.SidebarFilters:contains(VIEWING:)').should('be.visible');
+        cy.get('.SidebarFilters:contains(All channels)').should('be.visible');
+
+        // # Enable the unread filter
+        cy.get('.SidebarFilters_filterButton').should('be.visible').click();
+
+        // * Verify that the unread filter is in filter by unread state
+        cy.get('.SidebarFilters:contains(FILTERED BY:)').should('be.visible');
+        cy.get('.SidebarFilters:contains(Unread)').should('be.visible');
+
+        // # Disable the unread filter
+        cy.get('.SidebarFilters_filterButton').should('be.visible').click();
+
+        // * Verify that the unread filter is back in all channels state
+        cy.get('.SidebarFilters:contains(VIEWING:)').should('be.visible');
+        cy.get('.SidebarFilters:contains(All channels)').should('be.visible');
+    });
+
+    it('should only show unreads when the unread filter is enabled', () => {
```

Comment:
This test failed on me. PTAL
![Screen Shot 2020-03-18 at 8 38 24 PM](https://user-images.githubusercontent.com/5334504/76961909-32973e80-6959-11ea-9232-7449a2a7c1c5.png)